src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,21 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && !isset($file->data[$key]) && $member_name !== null) {
+                $traitAnnotations = array();
+
+                if (isset($file) && isset($file->traitMethodOverrides[$class_name][$member_name])) {
+                    list($traitName, $originalMemberName) = $file->traitMethodOverrides[$class_name][$member_name];
+                    $traitAnnotations = $this->getAnnotations($traitName, $member_type, $originalMemberName);
+                } else {
+                    foreach ($reflection->getTraitNames() as $traitName) {
+                        $traitAnnotations += $this->getAnnotations($traitName, $member_type, $member_name);
```

Comment:
This doesn't cover behavior with properties, that you've just discovered (property of 1st used trait is used if several used traits have it declared).

Pay attention here, because there can be a case, when both traits have a property, but only 2nd used trait has that property with annotations. That won't change the fact, that property without annotations from 1st used trait will be used.

There is no need to check if we're dealing with properties or methods here. Same logic applies. It's PHP who will end up in fatal error when both traits have same method declared.